### ISSUE TITLE
Greenlight frontend scales without load on the system

### ISSUE THREAD

**Issue by bklein1** [aws-samples/aws-scalable-big-blue-button-example#5]
**Describe the bug**
Greenlight frontend scales without load on the system

**To Reproduce**
1. Deploy the whole stack
2. Wait for it to be deployed
3. Monitor scaling activities for Greenlight and ECS Cluster

**Expected behavior**
No Scaling Activities should be present

**Comment by vsinghal339** [comment]
Hi,
I deploy the whole architecture in AWS once again and this issue still seems to be not resolved. It still scales without any load and the policy is still:

"As required to maintain metric CapacityProviderReservation (namespace: AWS/ECS/ManagedScaling, dimensions: CapacityProviderName = bbb-stack-BBBECSStack-1WETWFX1TQMNR-BBBECSCapacityProvider-tn5rc3502vgq, ClusterName = bbb-stack-BBBECSStack-1WETWFX1TQMNR-ECSCluster, statistic: Average, unit: ) at 85"

**Comment by bklein1** [comment]
Hi @vsinghal339,

Can you provide me some insights on the metrics that caused the scaling? What threshold was set? What was the actual metrics number?

**Comment by vsinghal339** [comment]
![Uploading Screenshot 2020-09-29 at 12.50.34 PM.png…]()

**Comment by vsinghal339** [comment]
For BBBAppECS the AutoScaling policy is 
"AutoScaling-ECSManagedAutoScalingPlan-3ce7ce8c-97db-4271-bd71-012d61d76d46-782d0528-55ba-4f00-a97c-38f9bbce0e4a

Policy type:
Target tracking scaling
Enabled or disabled?
Enabled
Execute policy when:
As required to maintain metric CapacityProviderReservation (namespace: AWS/ECS/ManagedScaling, dimensions: CapacityProviderName = bbb-stack-BBBECSStack-XSFDHUVSOD69-BBBECSCapacityProvider-U82uC8ShHAcq, ClusterName = bbb-stack-BBBECSStack-XSFDHUVSOD69-ECSCluster, statistic: Average, unit: ) at 85"

**Comment by vsinghal339** [comment]
I also noticed that Recordings are also not coming up now. Last night I ran destroy.sh file to removeall architecture and then re-run the setup.sh but now recordings are not coming up and ECS stacks are not coming down

**Comment by bklein1** [comment]
Can you please share the metrics of the Greenlight Service? The capacity provider scales based on the number of tasks provisioned to the cluster.
Can you please also provide the User Data that is used by the BBB Application server?

**Comment by vsinghal339** [comment]
Where can i find all those metrics?

**Comment by bklein1** [comment]
The metrics can be found under Cloudwatch --> Alarms. There you can filter for Greenlight. You should see 2 Alarms now. One in state Alarm and one in state OK. Click on the one with the condition: `RequestCountPerTarget` > 80 for 3 datapoints within 3 minutes` and check for the graph.

The User Data is part of the Instance Meta Data. You can either check it on one of the BBB Application instances or at the corresponding launch configuration.

**Comment by vsinghal339** [comment]
<img width="1792" alt="Screenshot 2020-09-29 at 1 37 12 PM" src="https://user-images.githubusercontent.com/20592184/94530663-07bb6a80-0259-11eb-8891-3020e158cda1.png">

**Comment by vsinghal339** [comment]
And the user data for Bigbluebutton:

#!/bin/bash -xe
apt update -y
DEBIAN_FRONTEND='noninteractive' apt -y -o Dpkg::Options::='--force-confdef' -o Dpkg::Options::='--force-confold' full-upgrade

apt autoremove -y
apt clean
apt autoclean

apt install -y git binutils python3-pip python-pip build-essential python-dev python-setuptools jq

pip install https://s3.amazonaws.com/cloudformation-examples/aws-cfn-bootstrap-latest.tar.gz
pip install -U awscli

cd /tmp
git clone https://github.com/aws/efs-utils
cd efs-utils
./build-deb.sh
apt-get -y install ./build/amazon-efs-utils*deb

# Set instance Hostname
instance_ipv4=$(curl -s http://169.254.169.254/latest/meta-data/public-ipv4)
instance_random=$(cat /dev/urandom | tr -dc 'a-z0-9' | fold -w 6 | head -n 1)
instance_publichostname=vc-$instance_random
instance_fqdn=$instance_publichostname.session.swye360.com

# register in route53
wget --tries=10 https://github.com/barnybug/cli53/releases/download/0.8.17/cli53-linux-amd64 -O /usr/local/bin/cli53
sudo chmod +x /usr/local/bin/cli53

# create script for route53-handler
aws s3 cp s3://bbb-stack-sources-bbbstackbucket-nuuotyhr3f6o/route53-handler.service /etc/systemd/system/route53-handler.service
aws s3 cp s3://bbb-stack-sources-bbbstackbucket-nuuotyhr3f6o/route53-handler.sh /usr/local/bin/route53-handler.sh
chmod +x /usr/local/bin/route53-handler.sh

sed -i "s/INSTANCE_PLACEHOLDER/$instance_publichostname/g" /etc/systemd/system/route53-handler.service
sed -i "s/ZONE_PLACEHOLDER/Z087539529L7NH3MEUBH6/g" /etc/systemd/system/route53-handler.service

systemctl daemon-reload
systemctl enable route53-handler
systemctl start route53-handler

mkdir -p /mnt/bbb-recordings
echo "fs-80aac685:/ /mnt/bbb-recordings efs defaults,tls,rw 0 0" >> /etc/fstab

mkdir -p /var/bigbluebutton

if test -e "/dev/nvme1n1"; then
  DEVICE=/dev/nvme1n1
  parted -s -a optimal -- $DEVICE mklabel gpt mkpart primary 1MiB -2048s
  sleep 20s
  mkfs.ext4 -F $DEVICE\p1
  UUID=$(blkid |grep "$DEVICE\p1" | awk '{print $2}' |sed 's/"//g')
else
  DEVICE=/dev/sdf
  parted -s -a optimal -- $DEVICE mklabel gpt mkpart primary 1MiB -2048s
  sleep 20s
  mkfs.ext4 -F $DEVICE\1
  UUID=$(blkid |grep "$DEVICE\1" | awk '{print $2}' |sed 's/"//g')
fi

echo "$UUID       /var/bigbluebutton   ext4    defaults,nofail        0       2" >> /etc/fstab
mount -a


turn_hostname=$(aws ssm get-parameter --region us-west-2 --name "turnhostname" --with-decryption --output text --query Parameter.Value)
turn_fqdn=$turn_hostname.session.swye360.com
turnsecret=$(aws secretsmanager get-secret-value --region us-west-2 --secret-id arn:aws:secretsmanager:us-west-2:849423638545:secret:BBBTurnSecret-SFiASnnYqh6A-oVijXa --query SecretString --output text | jq -r .turnkeyvalue)

sleep 1m

x=1
while [ $x -le 5 ]
do
  until host $instance_fqdn  | grep -m 1 "has address $instance_ipv4"; do sleep 5 ; done
  x=$(( $x + 1 ))
done

/usr/local/bin/cfn-signal -e $? --stack bbb-stack-BBBAppStack-11V8TAHWINCKO --resource BBBApplicationAutoScaling --region us-west-2 || true

wget -qO- https://bbb-install.s3.us-east-2.amazonaws.com/bbb-install-v2.sh | bash -s -- -v xenial-220 -s $instance_fqdn -e partners@swye360.com -c $turn_fqdn:$turnsecret

groupadd -g 2000 scalelite-spool
usermod -a -G scalelite-spool bigbluebutton

wget --tries=10  https://raw.githubusercontent.com/blindsidenetworks/scalelite/master/bigbluebutton/scalelite_post_publish.rb -O /usr/local/bigbluebutton/core/scripts/post_publish/scalelite_post_publish.rb
chmod +x /usr/local/bigbluebutton/core/scripts/post_publish/scalelite_post_publish.rb
aws s3 cp s3://bbb-stack-sources-bbbstackbucket-nuuotyhr3f6o/scalelite-config.yml /usr/local/bigbluebutton/core/scripts/scalelite.yml

# create script for scalelite-handler
aws s3 cp s3://bbb-stack-sources-bbbstackbucket-nuuotyhr3f6o/scalelite-handler.service /etc/systemd/system/scalelite-handler.service
aws s3 cp s3://bbb-stack-sources-bbbstackbucket-nuuotyhr3f6o/scalelite-handler.sh /usr/local/bin/scalelite-handler.sh
chmod +x /usr/local/bin/scalelite-handler.sh

SERVER="$(bbb-conf --secret | head -2 | tail -1 | sed -r 's/.*URL: //g')api"
SECRET=$(bbb-conf --secret | head -3 | tail -1 | sed -r 's/.*Secret: //g')

sed -i "s/SECRET_PLACEHOLDER/$SECRET/g" /etc/systemd/system/scalelite-handler.service
sed -i "s|SERVER_PLACEHOLDER|$SERVER|g" /etc/systemd/system/scalelite-handler.service
sed -i "s/AWSREGION_PLACEHOLDER/us-west-2/g" /etc/systemd/system/scalelite-handler.service
sed -i "s/ECSCLUSTER_PLACEHOLDER/bbb-stack-BBBECSStack-XSFDHUVSOD69-ECSCluster/g" /etc/systemd/system/scalelite-handler.service

systemctl daemon-reload
systemctl enable scalelite-handler
systemctl start scalelite-handler

# create script for turn-handler
aws s3 cp s3://bbb-stack-sources-bbbstackbucket-nuuotyhr3f6o/turn-handler.service /etc/systemd/system/turn-handler.service
aws s3 cp s3://bbb-stack-sources-bbbstackbucket-nuuotyhr3f6o/turn-handler.timer /etc/systemd/system/turn-handler.timer
aws s3 cp s3://bbb-stack-sources-bbbstackbucket-nuuotyhr3f6o/turn-handler.sh /usr/local/bin/turn-handler.sh
chmod +x /usr/local/bin/turn-handler.sh

sed -i "s/AWSREGION_PLACEHOLDER/us-west-2/g" /etc/systemd/system/turn-handler.service
sed -i "s|PARAMETER_PLACEHOLDER|turnhostname|g" /etc/systemd/system/turn-handler.service

systemctl daemon-reload
systemctl enable turn-handler.timer
systemctl start turn-handler.timer

curl https://s3.amazonaws.com/aws-cloudwatch/downloads/latest/awslogs-agent-setup.py -O
chmod +x ./awslogs-agent-setup.py
./awslogs-agent-setup.py -n -r us-west-2 -c s3://bbb-stack-sources-bbbstackbucket-nuuotyhr3f6o/bbb-cwagent.config
systemctl enable awslogs

**Comment by vsinghal339** [comment]
Any update on this?

**Comment by bklein1** [comment]
I am currently re-deploying the stack from scratch and try to reproduce the error. Will keep you updated as soon as I found something

**Comment by bklein1** [comment]
Hi @vsinghal339,

I deployed the whole stack again and I saw following behaviour:

1. ECS Cluster was started with 3 Nodes
2. Frontend Stack was deployed with following configuration:
    1. Greenlight having 3 Replicas
    2. Scalelight having 3 Replicas
3. ECS Capacity Provider initiates scaling of EC2 instances --> 4 Instances
4. No scaling of Greenlight or Scalelite was observed
5. No further scaling of EC2 Instances was observed

This is expected behaviour as of now and is caused by the Capacity Provider Reservation Alarm is set to < 85. There is https://github.com/aws-samples/aws-scalable-big-blue-button-example/issues/12 which tackles this inefficiency.

Did you observe different bahaviour?
